src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacNumericValidator.java

```diff
@@ -531,14 +545,14 @@ private QuantityValidationResponse exactStringMatch(final IsaacNumericQuestion i
                 }
 
                 if (questionAnswerString.toString().trim().equals(userStringForComparison.toString().trim())) {
-                    Boolean unitFeedback = null;
+                    Boolean unitsCorrect = null;
                     if (isaacNumericQuestion.getRequireUnits()) {
-                        unitFeedback = quantityFromQuestion.getUnits().equals(answerFromUser.getUnits());
+                        unitsCorrect = wasACorrectAnswerWithUsersSelectedUnit || quantityFromQuestion.isCorrect();
                     }
 
                     return new QuantityValidationResponse(isaacNumericQuestion.getId(), answerFromUser,
                             quantityFromQuestion.isCorrect(), (Content) quantityFromQuestion.getExplanation(),
-                            quantityFromQuestion.isCorrect(), unitFeedback, new Date());
+                            quantityFromQuestion.isCorrect(), unitsCorrect, new Date());
```

Comment:
Why not just use `wasACorrectAnswerWithUsersSelectedUnit` here? Or, better, rename that outer variable to `unitsCorrect` and ditch this inner variable entirely.